### ISSUE TITLE
Implement new TFP optimisers?

### ISSUE THREAD

**Issue by njtierney** [greta-dev/greta#568]
https://www.tensorflow.org/probability/api_docs/python/tfp/optimizer

* [differential evolution](https://www.tensorflow.org/probability/api_docs/python/tfp/optimizer/differential_evolution_minimize)
* [differential evolution one step?](https://www.tensorflow.org/probability/api_docs/python/tfp/optimizer/differential_evolution_one_step)
* [Hessian-informed proximal gradient descent](https://www.tensorflow.org/probability/api_docs/python/tfp/optimizer/proximal_hessian_sparse_minimize)
* [Hessian-informed proximal gradient descent](https://www.tensorflow.org/probability/api_docs/python/tfp/optimizer/proximal_hessian_sparse_one_step)
